core/cas-server-core-authentication-mfa-api/src/main/java/org/apereo/cas/authentication/MultifactorAuthenticationUtils.java

```diff
@@ -61,6 +64,20 @@ public static MultifactorAuthenticationProviderBypass newMultifactorAuthenticati
         return new HashMap<>(0);
     }
 
+    /**
+     * Method returns an Optional that will contain a MultifactorAuthenticationProvider that has the
+     * same id as the passed providerId parameter.
+     *
+     * @param providerId - the id to match
+     * @param context - ApplicationContext
+     * @return - Optional
+     */
+    public static Optional<MultifactorAuthenticationProvider> getMultifactorAuthenticationProviderById(final String providerId,
+                                                                                                       final ApplicationContext context) {
+        return getAvailableMultifactorAuthenticationProviders(context).values().stream()
+                .filter(p -> p.getId().equals(providerId)).findFirst();
```

Comment:
Do we care about case in equality?

Reply:
Not sure.  At least everywhere I have used it, the values are always sourced from the mfa properties configuration.  Changing to equalsIgnoreCase I think would be fine.